wikihow2zim/scraper.py

```diff
@@ -76,12 +76,161 @@ def add_assets(self):
+                )
+
+                text = s.find("div", {"class": "step"})
+                if text.find("script"):
+                    text.find("script").decompose()
+                if text.find("b", {"class": "whb"}):
+                    text.find("b", {"class": "whb"}).decompose()
+                sups = text.find_all("sup")
+                for ss in sups:
+                    ss.decompose()
+                step["text"] = text
+
+                method["steps"].append(step)
+
+            methods.append(method)
+
+        sys.exit()
+
+        # article = self.env.get_template("article.html")
+        # self.creator.add_item_for(
+        #     path="category/" + article_title,
+        #     title="article_title",
+        #     content=article.render(title=article_title, description=description),
+        #     mimetype="text/html",
+        # )
+
+        # Add images:
+        # url = (
+        #     "https://www.wikihow.com/extensions/wikihow/mobile"
+        #     "/images/wikihow_logo_230.png"
+        # )
+        # self.creator.add_item(
+        #     URLItem(
+        #         url=url,
+        #         path="assets/static/wikihow_logo.png",
+        #         mimetype="image/png",
+        #     )
+        # )
+
+    def walk_subcategories(self, main_wiki_url, cat_url, recursion_depth):
+        try:
+            response = requests.get(cat_url)
+        except Exception as exc:
+            logger.critical(f"Unable to retrieve homepage at {cat_url}: {exc}")
+            logger.exception(exc)
+
+        soup = bs4.BeautifulSoup(response.text, "html.parser")
+        article_divs = soup.find_all("div", {"class": "responsive_thumb"})
+        for article_div in article_divs:
+            article_url = article_div.find("a")["href"]
+            if not article_url in self.articles_list:
+                self.articles_list[article_url] = [str(recursion_depth) + " " + cat_url]
+                self.get_article(article_url)
+            else:
+                self.articles_list[article_url].append(
+                    str(recursion_depth) + " " + cat_url
+                )
+
+        subcat_divs = soup.find_all("div", {"class": "subcat_container"})
+        for subcat_div in subcat_divs:
+            subcat_url = main_wiki_url + subcat_div.find("a")["href"]
+            if not subcat_url in self.subcat_list:
+                self.subcat_list[subcat_url] = [str(recursion_depth) + " " + cat_url]
+            else:
+                self.subcat_list[subcat_url].append(
+                    str(recursion_depth) + " " + cat_url
+                )
+            print(str(recursion_depth) + " category " + subcat_url, flush=True)
+            self.walk_subcategories(main_wiki_url, subcat_url, recursion_depth + 1)
+
+    def walk_categories(self):
+        # Get all the articles by walking through all the main categories
+        # and the sub-categories
+
+        self.articles_list = {}  # keep track of each article processed and
```

Comment:
Don't create scraper instance properties inside methods if those are to be used anyway. This should be moved to __init__

Reply:
@rgaudin Let's Keep them for now. We can remove them later on.